### PR TITLE
NO-JIRA: Additional job name verification tests

### PR DESCRIPTION
**Pull request by stbenjam**
Guard against job names being incorrect for techpreview and FIPS jobs.

### CODE REVIEW

File: test/extended/ci/job_names.go [hunk 1]
```diff
@@ -33,6 +33,17 @@ var _ = g.Describe("[sig-ci] [Early] prow job name", func() {
 	}
 	isPeriodic := strings.HasPrefix(jobName, "periodic-")
 
+	g.It("should match feature set", func() {
+		if jobName == "" {
+			e2eskipper.Skipf("JOB_NAME env var not set, skipping")
+		}
+
+		isTechPreview := exutil.IsTechPreviewNoUpgrade(oc)
+		if strings.Contains(jobName, "-techpreview") && !isTechPreview {
```
Comment (deads2k) [inline test/extended/ci/job_names.go:42]: Check in the other direction too. If the job is tech review, it should be in the name. Right?

**Review by sosiouxme — APPROVED** [review]
/lgtm

**Pull request merged**